dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/trigramsummary/TrigramSummaryControllerTest.java

```diff
@@ -0,0 +1,254 @@
+        teaC = createTrackedEntityAttribute( 'C' );
+        teaD = createTrackedEntityAttribute( 'D' );
+        teaE = createTrackedEntityAttribute( 'E' );
+        teaF = createTrackedEntityAttribute( 'F' );
+        teaF.setUnique( true );
+        manager.save( teaA );
+        manager.save( teaB );
+        manager.save( teaC );
+        manager.save( teaD );
+        manager.save( teaE );
+        manager.save( teaF );
+
+        trackedEntityType = createTrackedEntityType( 'A' );
+        manager.save( trackedEntityType );
+
+        tetaA = new TrackedEntityTypeAttribute();
+        tetaA.setSearchable( true );
+        tetaA.setTrackedEntityType( trackedEntityType );
+        tetaA.setTrackedEntityAttribute( teaA );
+        manager.save( tetaA );
+        trackedEntityType.getTrackedEntityTypeAttributes().add( tetaA );
+
+        tetaB = new TrackedEntityTypeAttribute();
+        tetaB.setSearchable( false );
+        tetaB.setTrackedEntityType( trackedEntityType );
+        tetaB.setTrackedEntityAttribute( teaD );
+        manager.save( tetaB );
+        trackedEntityType.getTrackedEntityTypeAttributes().add( tetaB );
+
+        manager.update( trackedEntityType );
+
+        pteaA = new ProgramTrackedEntityAttribute();
+        pteaA.setSearchable( true );
+        pteaA.setAttribute( teaB );
+        pteaA.setProgram( program );
+        manager.save( pteaA );
+
+        pteaB = new ProgramTrackedEntityAttribute();
+        pteaB.setSearchable( false );
+        pteaB.setAttribute( teaE );
+        pteaB.setProgram( program );
+        manager.save( pteaB );
+
+        program.getProgramAttributes().add( pteaA );
+        program.getProgramAttributes().add( pteaB );
+        manager.update( program );
+
+        tei = createTrackedEntityInstance( orgUnit );
+        tei.setTrackedEntityType( trackedEntityType );
+        manager.save( tei );
+
+        programInstance = new ProgramInstance( program, tei, orgUnit );
+        programInstance.setAutoFields();
+        programInstance.setEnrollmentDate( new Date() );
+        programInstance.setIncidentDate( new Date() );
+        programInstance.setStatus( ProgramStatus.COMPLETED );
+        programInstance.setFollowup( true );
+        manager.save( programInstance );
+    }
+
+    @Test
+    void getTrigramIndexSummaryWhenNoIndexesAreCreated()
+    {
+
+        when( trackedEntityAttributeTableManager.getAttributeIdsWithTrigramIndexCreated() )
+            .thenReturn( new ArrayList<>() );
+        TrigramSummary trigramSummary = controller.getTrigramSummary( new HashMap<>() );
+
+        assertNotNull( trigramSummary );
+
+        Set<String> expected = Set.of( "\"AttributeA\"", "\"AttributeB\"", "\"AttributeF\"" );
+        assertAttributeList( trigramSummary.getIndexableAttributes(), expected );
+
+        Set<String> expected2 = Set.of();
+        assertAttributeList( trigramSummary.getIndexedAttributes(), expected2 );
+
+        Set<String> expected3 = Set.of();
```

Comment:
The names `expected2` `expected3` show that its hard to come up with a good name. Lucky for us, we dont to come up with a name. Just inline them 😄